### PR TITLE
chore: properly isolate module tests

### PR DESCRIPTION
**Pull request by Narretz**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

**What is the current behavior? (You can also link to an open issue here)**
Previously, our module tests weren't running in isolation. Theroetically, we wanted to test if the angular.js *build*  file works well with the individual module *source* files (Note that we don't use the module *build* files because we sometimes test non-public apis).
However, we bundled all module files together with the build file, and this messed up the results as some modules assign angular helpers like isDefined in the current scope, which means all following modules have access to them too, which essentially defeats the purpose of these tests. See #16702.

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?**

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our [guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [ ] Fix/Feature: [Docs](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#documentation) have been added/updated
- [ ] Fix/Feature: Tests have been added; existing tests pass

**Other information**:

### CODE REVIEW

File: angularFiles.js [hunk 1]
```diff
@@ -189,21 +189,78 @@ var angularFiles = {
     'src/angular.bind.js'
   ],
 
-  'karmaModules': [
+  'karmaModulesBase': [
     'build/angular.js',
-    '@angularSrcModules',
+    'src/ngMock/*.js',
```
Comment (gkalpak) [inline angularFiles.js]: Does this have to be here? Can't it shadow problems as before (e.g. modules depending on stuff that is defined in `ngMock`'s source code)?

File: angularFiles.js [hunk 2]
```diff
@@ -189,21 +189,78 @@ var angularFiles = {
...
+
+  'karmaModules-ngMessageFormat': [
+    '@karmaModulesBase',
+    '@angularSrcModuleNgMessageFormat',
+    'test/ngMessageFormat/**/*.js'
+  ],
+
+  'karmaModules-ngMessages': [
+    '@karmaModulesBase',
+    'build/angular-animate.js',
```
Comment (gkalpak) [inline angularFiles.js]: Hm... 🤔
Why is this necessary?

File: angularFiles.js [hunk 3]
```diff
@@ -189,21 +189,78 @@ var angularFiles = {
...
+    '@angularSrcModuleNgMessages',
+    'test/ngMessages/**/*.js'
+  ],
+
+  'karmaModules-ngMock': [
+    '@karmaModulesBase',
+    'build/angular-resource.js',
```
Comment (gkalpak) [inline angularFiles.js]: Hm... 🤔 
Why is this necessary?

File: angularFiles.js (hunk 2 above)
Comment (Narretz) [reply angularFiles.js]: Some tests in ngMessages are dependent on ngAnimate, e.g. https://github.com/angular/angular.js/blob/bb5a7e39ba2c8fe5afe1ec6f90ebedf22658a1c8/test/ngMessages/messagesSpec.js#L465

File: angularFiles.js (hunk 3 above)
Comment (Narretz) [reply angularFiles.js]: There's a test that uses ngResource: https://github.com/angular/angular.js/blob/bb5a7e39ba2c8fe5afe1ec6f90ebedf22658a1c8/test/ngMock/angular-mocksSpec.js#L1114
I think we could use a custom module here instead.

File: angularFiles.js (hunk 1 above)
Comment (Narretz) [reply angularFiles.js]: the angular-mocks file defines the module fn that integrates with jasmine, so I think it unfortunately must always be included: https://github.com/angular/angular.js/blob/bb5a7e39ba2c8fe5afe1ec6f90ebedf22658a1c8/src/ngMock/angular-mocks.js#L3085

File: angularFiles.js (hunk 2 above)
Comment (gkalpak) [reply angularFiles.js]: Do we still need `ngAnimate` if we use `ngAnimateMock`?

File: angularFiles.js (hunk 3 above)
Comment (gkalpak) [reply angularFiles.js]: 👍 for using a custom module.

File: angularFiles.js (hunk 1 above)
Comment (gkalpak) [reply angularFiles.js]: But we get the same integration with `build/angular-mocks.js`, right? (Without the problem of making things defined in the `ngMocks` closure available to other modules.)

I would expect `build/angular-mocks.js` to be included in all module tests, except for `ngMocks` specs, which should indeed use `src/ngMock/*.js`. Do I miss something?

File: angularFiles.js (hunk 2 above)
Comment (Narretz) [reply angularFiles.js]: Theoretically yes, but I'm more comfortable with testing the actual animations instead of just the base provided by ng.

File: angularFiles.js (hunk 1 above)
Comment (Narretz) [reply angularFiles.js]: Using `build/angular-mocks.js` works great, good suggestion!
In the beginning, it was a big hassle to separate all these files, it's possible that at some point the ngMock src files had to be included.
